### PR TITLE
.gemspecs are related to Gemfile.lock

### PR DESCRIPTION
**Pull request by kbarrette**

### CODE REVIEW

File: spec/bibliothecary_spec.rb [hunk 1]
```diff
@@ -85,13 +85,13 @@
...
         success: true,
-        :related_paths=>["Gemfile"]},
+        :related_paths=>["Gemfile", "bibliothecary.gemspec"]},
        {:platform=>"rubygems",
         :path=>"bibliothecary.gemspec",
         :dependencies=>[],
         :kind => 'manifest',
         success: true,
-        :related_paths=>[]}])
+        :related_paths=>["Gemfile.lock"]}])
```
Comment (havocp) [inline spec/bibliothecary_spec.rb]: I'm wondering if we want Gemfile to show up here too; but maybe figure out how we use it and it might not be necessary

Comment (kbarrette) [reply spec/bibliothecary_spec.rb]: It seems like we have manifest->lockfile and lockfile->manifest related paths, but not manifest->manifest, although I see what you're saying. I'll open another pull if I need to.

Comment (kbarrette) [reply spec/bibliothecary_spec.rb]: Although as I think about it more, you're probably right. Let me check our usage of this.

File: lib/bibliothecary/parsers/rubygems.rb [hunk 2]
```diff
@@ -50,7 +49,16 @@ def self.parse_gemspec(file_contents)
         parse_ruby_manifest(manifest)
       end
 
-
+      def self.set_related_paths_field(by_dirname_dest, by_dirname_source)
```
Comment (katzj) [inline lib/bibliothecary/parsers/rubygems.rb]: It's probably worth some comments about how this varies from the base case -- I'm also a little wary that we'll make a change to one and not the other as it's not clear that there's overriding happening here

Comment (kbarrette) [reply lib/bibliothecary/parsers/rubygems.rb]: Yeah, you're right - I'll rearrange this to be more declarative

**Comment by kbarrette** [comment]
@katzj @havocp updated - I couldn't resist a bit of refactoring, but I can pull it out if it'll slow down review.

File: lib/bibliothecary/parsers/rubygems.rb [hunk 3]
```diff
@@ -12,16 +12,18 @@ def self.mapping
...
             kind: 'manifest',
-            parser: :parse_gemfile
+            parser: :parse_gemfile,
+            related_to: [ 'manifest', 'lockfile' ]
           },
           match_extension(".gemspec") => {
             kind: 'manifest',
             parser: :parse_gemspec,
-            can_have_lockfile: false
+            related_to: [ 'manifest', 'lockfile' ]
```
Comment (kbarrette) [inline lib/bibliothecary/parsers/rubygems.rb]: Open to other ideas for how to indicate this, but this seemed the most obvious

**Pull request merged**